### ISSUE TITLE
Reduce number of Puppeteer installer dependencies

### ISSUE THREAD

**Issue by Krinkle** [puppeteer/puppeteer#5026]
For the vast functionality and joy that Puppeteer brings, it has a fairly small number of dependencies. Especially compared to other big players in the npm ecosystem, requiring only **43 packages** is an applaudable achievement.

I was even more delighted to find the following in the [contribution guidelines](https://github.com/GoogleChrome/puppeteer/blob/v1.20.0/CONTRIBUTING.md#adding-new-dependencies):

> * Do not add a new dependency if the desired functionality is easily implementable.
> * Do not add a new dependency unless it's critical to project success.
> …
> * If adding a dependency, it should be well-maintained and trustworthy.

I'm filing this issue as explicit objective, to see what we can do here and/or in upstream projects we depend on, to bring this number down even further. Tentatively aiming for 20, although under 10 would be best.

Related reading:
* https://github.com/qunitjs/qunit/issues/1342
* https://timotijhof.net/posts/2019/protect-yourself-from-npm/

| [puppeteer-core@1.20.0   plotted on npm.avanka.com](https://npm.anvaka.com/#/view/2d/puppeteer-core/1.20.0) _(as of 10 Oct 2019)_
|--
| <img width="1098" alt="Screenshot" src="https://user-images.githubusercontent.com/156867/66615297-04504e80-ebc4-11e9-9e68-5f5db8c9fbc0.png">

**Comment by Krinkle** [comment]
Some low-hanging fruit perhaps, the `debug` package is shipped three times:

* `debug@4.x` from puppeteer@1.20.
* `debug@3.x` from https-proxy-agent@2.2.2.
* `debug@2.x` from extract-zip@1.6.7.

**Comment by VictorGaiva** [comment]
A good reason to take a deeper look into this:
```
$ npm audit
                                                                                
                       === npm audit security report ===                        
                                                                                
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Machine-In-The-Middle                                        │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ https-proxy-agent                                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.0.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ puppeteer                                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ puppeteer > https-proxy-agent                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/1184                            │
└───────────────┴──────────────────────────────────────────────────────────────┘
found 1 high severity vulnerability in 2510 scanned packages
  1 vulnerability requires manual review. See the full report for details.
```

**Comment by Larry1123** [comment]
https://github.com/maxogden/extract-zip/issues/71
It seems that extract-zip would not match the `well-maintained` part, currently.
